contracts/treasury/src/contract.rs

```diff
@@ -28,10 +28,12 @@ pub fn instantiate(
     deps: DepsMut<'_>,
     _env: Env,
     _info: MessageInfo,
-    _msg: InstantiateMsg,
+    msg: InstantiateMsg,
 ) -> ContractResult<CwResponse> {
     versioning::initialize(deps.storage, version!(CONTRACT_STORAGE_VERSION))?;
 
+    try_configure_reward_dispatcher(deps.storage, msg.rewards_dispatcher)?;
```

Comment:
is it doable to validate the rewards_dispatcher address as we do on `sudo`, `platform::contract::validate_addr(&deps.querier, &rewards_dispatcher)?;` ? Check pls, the order of contract instantiations.

Reply:
We'd have to sacrifice either of the checks in the contract instantiations because `dispatcher` also checks the validity of the `treasury` address.

Reply:
ok, let's keep it as it is